src/InertiaVisual_TEST.cc

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include <gtest/gtest.h>
+#include <string>
+
+#include <ignition/common/Console.hh>
+
+#include "test_config.h"  // NOLINT(build/include)
+
+#include "ignition/rendering/InertiaVisual.hh"
+#include "ignition/rendering/RenderEngine.hh"
+#include "ignition/rendering/RenderingIface.hh"
+#include "ignition/rendering/Scene.hh"
+
+using namespace ignition;
+using namespace rendering;
+
+class InertiaVisualTest : public testing::Test,
+                        public testing::WithParamInterface<const char *>
+{
+  /// \brief Test basic API
+  public: void InertiaVisual(const std::string &_renderEngine);
+};
+
+/////////////////////////////////////////////////
+void InertiaVisualTest::InertiaVisual(const std::string &_renderEngine)
+{
+  RenderEngine *engine = rendering::engine(_renderEngine);
+  if (!engine)
+  {
+    igndbg << "Engine '" << _renderEngine
+              << "' is not supported" << std::endl;
+    return;
+  }
+
+  ScenePtr scene = engine->CreateScene("scene");
+
+  // create visual
+  InertiaVisualPtr inertiaVisual = scene->CreateInertiaVisual();
+  ASSERT_NE(nullptr, inertiaVisual);
+
+  // check initial values
+  EXPECT_EQ(nullptr, inertiaVisual->BoxVisual());
+
+  ignition::math::Pose3d p(0.0, 1.0, 2.5, 1.0, 0.4, 0.4);
+  ignition::math::Vector3d s(0.5, 0.5, 0.5);
+  inertiaVisual->Load(p, s);
+
+  EXPECT_NE(nullptr, inertiaVisual->BoxVisual());
```

Comment:
can you expand the test to include other APIs like `Material` and `OgreObject`?

Reply:
I think the methods to get `Material` and `OgreObject` are local to Ogre 1 and 2 classes only, similar to the [LightVisual test](https://github.com/ignitionrobotics/ign-rendering/blob/ign-rendering5/src/LightVisual_TEST.cc#L54-L67) and the [Ogre1 LightVisual class](https://github.com/ignitionrobotics/ign-rendering/blob/ign-rendering5/ogre/src/OgreLightVisual.cc#L62-L68)